packages/typespec-vscode/src/tsp-language-client.ts

```diff
@@ -88,6 +95,23 @@ export class TspLanguageClient {
     }
   }
 
+  public async compileProject(
+    doc: TextDocumentIdentifier,
+    options?: CompilerOptions,
+  ): Promise<CustomCompileResult | undefined> {
+    const compileProjectRequestName: CustomRequestName = "typespec/internalCompile";
+    try {
+      const result = await this.client.sendRequest<CustomCompileResult>(compileProjectRequestName, {
+        doc: doc,
+        options: { ...options, dryRun: false },
```

Comment:
why do we need to set the "dryRun" to false explicitly here?

Reply:
Yes. "dryRun" default is true. and if "dryRun" is true, the code will not emit.